src/services/reminder.service.js

```diff
@@ -1,97 +1,106 @@
   const ReminderModel = require('../models/reminder.model.js');
-  const SubreminderService = require('../services/subreminder.service.js');
-  const ValidationUtils = require('../utils/validation.util.js');
+  const { MyDate } = require('../utils');
 
-  function findAllReminders() {
-    return ReminderModel.find().populate('event');
-  }
-
-  function findRemindersByCourseId(courseId) {
-    const course = new mongoose.Types.ObjectId(courseId);
-    return ReminderModel.find({ course });
-  }
-
-  function createReminders(remindersToCreate) {
+  function createReminder(config) {
     return new Promise((resolve, reject) => {
-      const createPromises = remindersToCreate.map(reminderConfig => {
-        try {
-          return createReminderWithConfig(reminderConfig);
-        } catch (err) {
-          console.error(
-            `Failed to create reminder with config "${reminderConfig}": `,
-            err
-          );
-        }
+      const { name, startDate, endDate, startTime, endTime, repeats } = config;
+      const reminders = createReminders(
+        startDate,
+        endDate,
+        repeats,
+        startTime,
+        endTime
+      );
+      const reminderModels = reminders.map(myDate => {
+        const dateTime = myDate.getDateObject();
+        return new Promise((resolve, reject) => {
+          const newRmd = new ReminderModel({
+            name,
+            dateTime
+          });
+          newRmd
+            .save()
+            .then(resolve)
+            .catch(reject);
+        });
       });
-      return Promise.all(createPromises)
-        .then(newReminders =>
-          newReminders.filter(reminder => reminder !== undefined)
-        )
-        .then(newReminders => {
-          resolve(newReminders);
-        })
-        .catch(reject);
-    });
-  }
 
-  function createReminder(name, courseId, startDate) {
-    return new Promise((resolve, reject) => {
-      try {
-        ValidationUtils.notNullOrEmpty(name, 'name');
-        const course = new mongoose.Types.ObjectId(courseId);
-        const newReminder = new ReminderModel({ name, course, startDate });
-        newReminder
-          .save()
-          .then(resolve)
-          .catch(reject);
-      } catch (err) {
-        reject(err);
-      }
+      Promise.all(reminderModels)
+        .then(values => {
+          values = values.map(value => value._id);
+          resolve(values);
+        })
+        .catch(err => {
+          reject(err);
+          console.log(err);
+        });
     });
   }
 
-  function deleteReminder(reminderId) {
-    return new Promise((resolve, reject) => {
-      try {
-        ValidationUtils.notNullOrEmpty(reminderId, 'reminderId');
-        ReminderModel.deleteOne({ _id: reminderId })
-          .then(resolve)
-          .catch(reject);
-      } catch (err) {
-        reject(err);
+  function createReminders(startDate, endDate, repeats, startTime, endTime) {
+    const startDateSplit = startDate.split('-');
```

Comment:
null checks

Reply:
Also, a one-time reminder does not have an endDate. e.g., "YES (e.g., an exam date)"